### ISSUE TITLE
Help?

### ISSUE THREAD

**Issue by osmooth** [thecodemaiden/alljoyn-iot-pi#1]
Hi I'm new to Alljoyn how do I make this CPP application using a default Alljoyn (AJCL) setup, I'm confused where all the libraries exist? please can you help?

**Comment by thecodemaiden** [comment]
This was written for Debian Linux (Raspbian), and it assumes you have already installed AllJoyn to /usr/local

**Comment by osmooth** [comment]
Sorry yes I'm using Raspian, I'm totally confused with the folders and can't find an ideal way to install alljoyn, although I've downloaded the latest repo from their git, I can run the about and other samples from their samples folder. How can I install to /usr/local ? sorry it's a noob question but I'm totally stranded.

**Comment by thecodemaiden** [comment]
I'm sorry, I'm going to be very busy for the next couple of weeks.

It's been a while since I installed AllJoyn, and I don't have a Pi to sacrifice right now, so these instructions may not be 100% accurate. You can look up documentation on Makefile variables (like CFLAGS, CXXFLAGS, LIBS, INCLUDE) to make sure.

You can try to compile this by comparing the Makefile to the one in the sample directory.

Add a line under the INCLUDE = /usr/local/include like

`
INCLUDE += /path/to/sample/directory/headers
`
and another line

`
LDFLAGS = -L/path/to/built/libraries
`

and change the ROUTER_OBJ line to the path where the BundledRouter.o got built.

Then you need to refer to LDFLAGS in the build rules:

```
led_service:    led_service.o
    $(CXX) -o $@ led_service.o $(ROUTER_OBJ) $(LDFLAGS) $(LIBS) -lwiringPi

...

led_client: led_client.o
    $(CXX) -o $@ led_client.o $(ROUTER_OBJ) $(LDFLAGS) $(LIBS)
```

Try that and let me know if it builds.

**Comment by osmooth** [comment]
Hi, many thanks for the help, really appreciate it. Now I got past the compilation error by changing two things in the Makefile, I passed the includes in the CXXFLAGS, so that it's explicit. Also on my gen 1 pi I was getting a memory issue, so I changed the C++ compiler to C++11, I don't know how this resolved the problem but I need to do more research on this one. The ajrouter was giving me an issue so I changed this to use libajrouter.o. This allows me to compile and generate the executable, but it's not working. So I've parked this and started to build a sample program on the target to see how I can initiate the router and bus attachment etc.
